src/sinks/clickhouse/parse.rs

```diff
@@ -0,0 +1,219 @@
+        parse_int64,
+        parse_float32,
+        parse_float64,
+        parse_string,
+        parse_uuid,
+        parse_date,
+        parse_ipv4,
+        parse_ipv6,
+    ))(s)
+}
+
+
+fn parse_array(s: &str) -> IResult<&str, SqlType> {
+    let (rest, v) = preceded(
+        tag("Array"), 
+        delimited(tag("("), parse_sql_type, tag(")"))
+    )(s)?;
+    Ok((rest, SqlType::Array(v.into())))
+}
+
+fn parse_map(s: &str) -> IResult<&str, SqlType> {
+    let (rest, (k, v)) = preceded(
+        tag("Map"),
+        delimited(
+            tag("("), 
+            pair(parse_sql_type, preceded(tag(","), parse_sql_type)), 
+            tag(")"))
+    )(s)?;
+    Ok((rest, SqlType::Map(k.into(), v.into())))
+}
+
+fn parse_nullable(s: &str) -> IResult<&str, SqlType> {
+    let (rest, v) = preceded(
+        tag("Nullable"),
+        delimited(tag("("), parse_nullable_inner, tag(")"))
+    )(s)?;
+    Ok((rest, SqlType::Nullable(v.into())))
+}
+
+fn parse_nullable_inner(s: &str) -> IResult<&str, SqlType> {
+    alt((
+        parse_datetime64,
+        parse_datetime,
+        parse_static_type,
+        parse_fixed_string,
+        parse_decimal,
+    ))(s)
+}
+
+fn parse_datetime(s: &str) -> IResult<&str, SqlType> {
+    let (rest, _) = tag("DateTime")(s)?;
+    Ok((rest, SqlType::DateTime(DateTimeType::DateTime32)))
+}
+
+fn parse_datetime64(s: &str) -> IResult<&str, SqlType> {
+    let (rest, (precision, tz)) = preceded(
+        tag("DateTime64"),
+        delimited(
+            tag("("), 
+            pair(
+                p_u32, 
+                preceded(
+                    tag(","), 
+                    map_res(take_until1(")"), |s: &str| s.parse::<Tz>())
+                )
+            ), 
+            tag(")")
+        )
+    )(s)?;
+    Ok((rest, SqlType::DateTime(DateTimeType::DateTime64(precision, tz))))
+}
+
+fn parse_fixed_string(s: &str) -> IResult<&str, SqlType> {
+    let (rest, n) = preceded(
+        tag("FixedString"),
+        delimited(
+            tag("("), 
+            p_u64, 
+            tag(")")
+        ),
+    )(s)?;
+    Ok((rest, SqlType::FixedString(n as usize)))
+}
+
+fn parse_decimal(s: &str) -> IResult<&str, SqlType> {
+    let (rest, (p, s)) = preceded(
+        tag("Decimal"),
+        delimited(
+            tag("("), 
+            pair(
+                p_u8, 
+                preceded(
+                    tag(","),
+                    p_u8,
+                ),
+            ), 
+            tag(")"))
+    )(s)?;
+    Ok((rest, SqlType::Decimal(p, s)))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use chrono_tz::Tz;
+    use clickhouse_rs::types::{SqlType,DateTimeType};
+
+    #[test]
+    fn test_parse_datetime64() {
+        let table = vec![
+            ("DateTime64(3,Asia/Shanghai)", SqlType::DateTime(DateTimeType::DateTime64(3, Tz::Asia__Shanghai))),
+        ];
+        for (s, expect) in table {
+            let (_, actual) = parse_datetime64(s).unwrap();
```

Comment:
Would it be useful to also assert that the remainder string (the first element of the tuple) is empty?